tests/core/test_lightning_module.py

```diff
@@ -399,3 +399,12 @@ def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_va
         match=r"gradient_clip_algorithm='norm'\)` and have passed `clip_gradients\(gradient_clip_algorithm='foo'",
     ):
         trainer.fit(model)
+
+
+def test_core_lightning_to_core_module_migration():
+    """Test for checking whether LightningModule is being imported from `core.module`, even if `core.module` is
+    used it will be rediredted to `core.module`."""
+    from pytorch_lightning.core.module import LightningModule as lm1
+    from pytorch_lightning.core.module import LightningModule as lm2
```

Comment:
Is this a correct test? It imports exactly the same thing twice

Reply:
the test isn't required in the PR.. removed.

Reply:
@otaj thanks for pointing it out, the issue was, someone might have substituted all the core.lightning to core module. 

this was the original test 
```python3
def test():
    from pytorch_lightning.core.module import LightningModule as lm1
    from pytorch_lightning.core.lightning import LightningModule as lm2
    
    assert lm1.__module__ == lm2.__module__
```
cc @rohitgr7

Reply:
but we will be adding back lightning.py in another PR after this one. There it will make sense to add such a test

Reply:
@rohitgr7 I think we might need the test until we completely remove `core.lightning`

Reply:
@rohitgr7 thanks for clarification